dataset/reduction.cpp

```diff
@@ -4,12 +4,20 @@
 /// @author Simon Heybrock
 #include "scipp/common/reduction.h"
 #include "dataset_operations_common.h"
+#include "scipp/core/element/util.h"
 #include "scipp/dataset/reduction.h"
-#include "scipp/variable/reduction.h"
+#include "scipp/dataset/special_values.h"
 
 using scipp::common::reduce_all_dims;
 
 namespace scipp::dataset {
+namespace {
+void validate_nanmean(const VariableConstView &var) {
+  if (isInt(var.dtype()))
+    throw except::TypeError(
+        "nanmean on integer input variables is not supported. Use mean");
+}
```

Comment:
Numpy does accepts `int` in `nanmean`. Why throw instead of just forwarding to `mean` (or making `nanmean` work)?

Reply:
I had thought about that, and maybe that is best. I will change.